libpod/options.go

```diff
@@ -37,68 +35,146 @@ const (
 // WithStorageConfig uses the given configuration to set up container storage
 // If this is not specified, the system default configuration will be used
 // instead
-func WithStorageConfig(config *storage.StoreOptions) RuntimeOption {
-	return runtimeNotImplemented
+func WithStorageConfig(config storage.StoreOptions) RuntimeOption {
+	return func(rt *Runtime) error {
+		if rt.valid {
+			return fmt.Errorf("runtime has already been finalized")
+		}
+
+		rt.config.StorageConfig.RunRoot = config.RunRoot
+		rt.config.StorageConfig.GraphRoot = config.GraphRoot
+		rt.config.StorageConfig.GraphDriverName = config.GraphDriverName
+
+		rt.config.StorageConfig.GraphDriverOptions = make([]string, len(config.GraphDriverOptions))
+		copy(rt.config.StorageConfig.GraphDriverOptions, config.GraphDriverOptions)
+
+		rt.config.StorageConfig.UIDMap = make([]idtools.IDMap, len(config.UIDMap))
+		copy(rt.config.StorageConfig.UIDMap, config.UIDMap)
+
+		rt.config.StorageConfig.GIDMap = make([]idtools.IDMap, len(config.UIDMap))
+		copy(rt.config.StorageConfig.GIDMap, config.GIDMap)
+
+		return nil
+	}
 }
 
 // WithImageConfig uses the given configuration to set up image handling
 // If this is not specified, the system default configuration will be used
 // instead
 func WithImageConfig(defaultTransport string, insecureRegistries, registries []string) RuntimeOption {
-	return runtimeNotImplemented
+	return func(rt *Runtime) error {
+		if rt.valid {
+			return fmt.Errorf("runtime has already been finalized")
+		}
+
+		rt.config.ImageDefaultTransport = defaultTransport
+
+		rt.config.InsecureRegistries = make([]string, len(insecureRegistries))
+		copy(rt.config.InsecureRegistries, insecureRegistries)
+
+		rt.config.Registries = make([]string, len(registries))
+		copy(rt.config.Registries, registries)
+
+		return nil
+	}
 }
 
 // WithSignaturePolicy specifies the path of a file which decides how trust is
 // managed for images we've pulled.
 // If this is not specified, the system default configuration will be used
 // instead
 func WithSignaturePolicy(path string) RuntimeOption {
-	return runtimeNotImplemented
+	return func(rt *Runtime) error {
+		if rt.valid {
+			return fmt.Errorf("runtime has already been finalized")
```

Comment:
You've this same error verbiage in four or five functions.  Can you tweak them to at least be slightly different for each?

Reply:
I think the error should probably remain the same (given it's the same error condition everywhere), but you make a good point that it's shared everywhere - I'll make it a variable to remove the duplication